pkg/remoteconfig/state/repository_test.go

```diff
@@ -227,6 +350,19 @@ func TestUpdateWithIncorrectlySignedTargets(t *testing.T) {
 	assert.EqualValues(t, 0, state.TargetsVersion)
 	assert.EqualValues(t, 1, state.RootsVersion)
 	assert.Nil(t, state.OpaqueBackendState)
+
+	// The unverified repository shouldn't reject this on account of the targets signature because
+	// it shouldn't even look at that.
+	updatedProducts, err = ta.unverifiedRepository.Update(update)
+	assert.NoError(t, err)
+	assert.Equal(t, 1, len(updatedProducts))
+	state, err = ta.unverifiedRepository.CurrentState()
+	assert.NoError(t, err)
+	assert.Equal(t, 1, len(state.Configs))
+	assert.Equal(t, 1, len(state.CachedFiles))
+	assert.EqualValues(t, 1, state.TargetsVersion)
+	assert.EqualValues(t, 0, state.RootsVersion) // 0 because we don't start with the open source root
+	assert.Equal(t, testOpaqueBackendStateContents, state.OpaqueBackendState)
```

Comment:
<!-- Praises highlight something positive. Try to leave at least one of these comments per review (if it exists :^)  -->
**👏 Praise:** ‏Nice to test this case (and all the others of course)